### DISCUSSION TITLE
MQTT Commands not Working

### DISCUSSION THREAD

**Discussion by Dudesky1325** [mrlt8/docker-wyze-bridge#857] [Q&A]
This is literally the only complaint I have about this project, everything else has been improving with each update. 

For whatever reason, I can't get any of the MQTT commands to work. No matter what topic I try to publish to, I get "0" as each response, even to "get" commands. I had some rudimentary camera controls set up in Home Assistant but they used the old MQTT commands. 

Not sure what I'm doing wrong here but I'm likely making some noob mistake. Any help is appreciated.

**Comment by mrlt8** [comment]
Can you post the topic and payload you're trying to send?

0 usually indicates an error. Do you see anything in the logs? It should printout more information about what is happening.

**Comment by spuggy1968** [comment]
I tried sending a command to my Dog Cam through MQTT based on instructions.   I simply used the default examples to get an idea how this worked, and nothing happened, but when I checked the log, this is what I can see:

[dog-cam-v2] [CONTROL] Attempting to SET: rotary_degree="-90,0"
Exception in thread dog-cam-v2_control:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/app/wyzebridge/wyze_control.py", line 165, in camera_control
    resp = send_tutk_msg(sess, cmd)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/wyzebridge/wyze_control.py", line 203, in send_tutk_msg
    tutk_msg, topic, payload, payload_str = lookup_cmd(cmd, log)
                                            ^^^^^^^^^^^^^^^^^^^^
  File "/app/wyzebridge/wyze_control.py", line 271, in lookup_cmd
    return tutk_msg(*payload), topic, payload, payload_str
           ^^^^^^^^^^^^^^^^^^
TypeError: K11000SetRotaryByDegree.__init__() missing 2 required positional arguments: 'horizontal' and 'vertical'
[WyzeBridge] [MQTT] {'status': 'error', 'command': 'rotary_degree', 'payload': '"-90,0"', 'response': 'timed out'}

Any thoughts or idea what is happening.

I sent a command the night before, and the camera did move, but I was using the PTZ command but I did not know what I was doing, but the camera started to spin.   It was late, I went to bed, and the next day I tried, and nothing seems to work.

**Reply by mrlt8** [reply]
My bad. You don't need to wrap the payload in quotes.

Can you try setting the payload without quotes like:

```
-90,0
```

**Reply by spuggy1968** [reply]
I tried both ways yesterday, and it did not work.   Today it works perfectly!  Thanks for super fast response!

**Reply by mrlt8** [reply]
FYI, the `rotary_degree` topic also supports a few preset payloads like up, down, left, right.

**Reply by Dudesky1325** [reply]
Inputting only numerical values for rotary_degree or rotary_action will work for me. I have also noticed that the camera feed will go down a lot more often if too many commands are sent in quick succession.
